backend/inspirehep/records/api/literature.py

```diff
@@ -317,6 +318,31 @@ def add_figures(self, figures):
     def add_file(
         self, url, original_url=None, key=None, filename=None, *args, **kwargs
     ):
+        if current_s3_instance.is_s3_url(url) and not current_app.config.get(
+            "FORCE_PROCESS_S3_FILES", False
+        ):
+            result = {
```

Comment:
During the migration, we use the afs cache from redis to replace the afs url with the corresponding s3 url. Therefore, here you can get a file with an s3 url but without the correct key (the hash of the content) and we also still need the logic of having the key moved to the filename.

Reply:
:+1:

Reply:
I just realized this should have been caught by a test. We have a test checking the usage of the file cache but it isn't checking all fields in the file. Now that we have different behavior for when the url is an s3 url, it would be good to assert also the other fields in the afs cache test. Check out ```test_migrate_record_from_mirror_uses_local_cache_for_afs_files```